pkg/client/patch.go

```diff
@@ -148,7 +164,21 @@ func MergeFromWithOptions(obj runtime.Object, opts ...MergeFromOption) Patch {
 	for _, opt := range opts {
 		opt.ApplyToMergeFrom(options)
 	}
-	return &mergeFromPatch{from: obj, opts: *options}
+	return &mergeFromPatch{patchType: types.MergePatchType, createPatch: createMergePatch, from: obj, opts: *options}
+}
+
+// StrategicMergeFrom creates a Patch that patches using the strategic-merge-patch strategy with the given object as base.
+func StrategicMergeFrom(obj runtime.Object) Patch {
+	return &mergeFromPatch{patchType: types.StrategicMergePatchType, createPatch: createStrategicMergePatch, from: obj}
+}
+
+// StrategicMergeFromWithOptions creates a Patch that patches using the strategic-merge-patch strategy with the given object as base.
+func StrategicMergeFromWithOptions(obj runtime.Object, opts ...MergeFromOption) Patch {
```

Comment:
In my opinion, there should be just one constructor `StrategicMergeFrom` and it should have the variadic `opts`.

Reply:
Fair enough, done.
PTAL :)